src/wallet/test/wallet_tests.cpp

```diff
@@ -547,18 +547,11 @@ static int64_t AddTx(CWallet& wallet, uint32_t lockTime, int64_t mockTime, int64
         block = inserted.first->second;
         block->nTime = blockTime;
         block->phashBlock = &hash;
+        wallet.AddTransaction(MakeTransactionRef(tx), hash, 0);
+    } else {
+        wallet.AddTransaction(MakeTransactionRef(tx));
     }
-
-    CWalletTx wtx(&wallet, MakeTransactionRef(tx));
-    if (block) {
-        wtx.SetMerkleBranch(block, 0);
-    }
-    {
-        LOCK(cs_main);
```

Comment:
In commit "Remove CWalletTx merging logic from AddToWallet",

Should stay? Related to #12681.

Looks like [this code](https://github.com/bitcoin/bitcoin/pull/12681/files#diff-b2bb174788c7409b671c46ccc86034bdR3816) is not hit, otherwise the lock held assertion in `LookupBlockIndex` would cause travis failure here?